### DISCUSSION TITLE
FIlter FFB

### DISCUSSION THREAD

**Discussion by rodrigobaronio** [Ultrawipf/OpenFFBoard#98] [Ideas]
hello

I didn't find a filter in ffb, cases like the iracing that the ffb is generated at 60hz it gets very "grooved" the effect, other softwares have a filter in the ffb, which seems to emulate an effect of 360 hz or more, like simucube, calls reconstruction filter, VRS and EMC has this filter too, I didn't find it in openffb, do you have or do you have an idea to implement it?

**Comment by Ultrawipf** [comment]
There are several filters implemented in the OpenFFBoard software. What you need is the "Constant force filter" which can be set up in the GUI. This filter applies only to the constant force effect which games usually use for the main physics based effects and will be equivalent to the other implementations you have mentioned.
Internally effects are calculated at 1khz so using a lowpass on the constant force effect has an interpolating effect.
![CF Filter](https://github.com/Ultrawipf/OpenFFBoard/assets/1668112/3c10932a-eac3-44fe-bdd5-196bf3c9a888)

The other effects are internally filtered depending on encoder settings.
If using the TMC4671 there is also the option to filter all forces with a 25khz sampled filter but this is intended to tune the overall motor response.
